File: tuto1/catkin_ws/src/my_robot_tutorials/src/subscriber.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void callback_data(const std_msgs::String& msg){
	ROS_INFO("Message received : %s", msg.data.c_str());
}

int main(int argc, char **argv){
	ros::init(argc, argv, "cpp_robot_receiver");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("/robot_transmitter", 1000, callback_data);
	ros::spin();
}

File: tuto1/catkin_ws/src/my_robot_tutorials/scripts/subscriber.py
#!/usr/bin/env python2

import rospy
from std_msgs.msg import String

def callback_data(msg):
	rospy.loginfo("Message received: ")
	rospy.loginfo(msg)

if __name__ == '__main__':
	rospy.init_node('robot_receiver')
	sub = rospy.Subscriber("/robot_transmitter", String, callback_data)

	rate = rospy.spin()
